src/client/datascience/kernel-launcher/localKernelSpecFinderBase.ts

```diff
@@ -148,66 +147,7 @@ export abstract class LocalKernelSpecFinderBase {
         interpreter?: PythonEnvironment,
         cancelToken?: CancellationToken
     ): Promise<IJupyterKernelSpec | undefined> {
-        // This is a backup folder for old kernels created by us.
-        if (specPath.includes(oldKernelsSpecFolderName)) {
-            return;
-        }
-        let kernelJson: ReadWrite<IJupyterKernelSpec>;
-        try {
-            traceVerbose(
-                `Loading kernelspec from ${getDisplayPath(specPath)} for ${getDisplayPath(interpreter?.path)}`
-            );
-            kernelJson = JSON.parse(await this.fs.readLocalFile(specPath));
-        } catch {
-            traceError(`Failed to parse kernelspec ${specPath}`);
-            return undefined;
-        }
-        if (cancelToken?.isCancellationRequested) {
-            return undefined;
-        }
-
-        // Special case. If we have an interpreter path this means this spec file came
-        // from an interpreter location (like a conda environment). Modify the name to make sure it fits
-        // the kernel instead
-        kernelJson.name = interpreter ? getInterpreterKernelSpecName(interpreter) : kernelJson.name;
-
-        // Update the display name too if we have an interpreter.
-        const isDefaultPythonName = kernelJson.display_name.toLowerCase().match(isDefaultPythonKernelSpecSpecName);
-        if (!isDefaultPythonName && kernelJson.language === PYTHON_LANGUAGE && kernelJson.argv.length > 2) {
-            // Default kernel spec argv for Python kernels is `"python","-m","ipykernel_launcher","-f","{connection_file}"`
-            // Some older versions had `ipykernel` instead of `ipykernel_launcher`
-            // If its different, then use that as an identifier for the kernel name.
-            const argv = kernelJson.argv
-                .slice(1) // ignore python
-                .map((arg) => arg.toLowerCase())
-                .filter((arg) => !['-m', 'ipykernel', 'ipykernel_launcher', '-f', '{connection_file}'].includes(arg));
-            if (argv.length) {
-                kernelJson.name = `${kernelJson.name}.${argv.join('#')}`;
-            }
-        }
-        kernelJson.display_name =
-            kernelJson.language === PYTHON_LANGUAGE && isDefaultPythonName
-                ? interpreter?.displayName || kernelJson.display_name
-                : kernelJson.display_name;
```

Comment:
Note: I removed this code, we're no longer updating the display_name of the kernelspec.
We don't need this (we load kernelspecs from disc and give them our own display names in the controllers).

Also, this preserves the display names the users may have added for kernelspecs (instead of us blowing it away)